scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/definition/fragment/FragmentParameterTypingParser.scala

```diff
@@ -8,7 +9,24 @@ import scala.util.Try
 
 class FragmentParameterTypingParser(classLoader: ClassLoader, classDefinitions: Set[ClassDefinition]) {
 
+  private val typingResultCache: Cache[String, Try[TypingResult]] =
```

Comment:
I wonder, maybe we should move cache and parsing Special object like: ClassTypingResult? And use it everywhere where we do converting class to typing ?

Reply:
This PR changed a lot since that comment. There is now auto-refreshable, cache on the entire UIDefinitions service.